disco/types/message.py

```diff
@@ -328,6 +330,14 @@ def add_field(self, *args, **kwargs):
         """
         self.fields.append(MessageEmbedField(*args, **kwargs))
 
+    def to_dict(self, iso=False):
```

Comment:
I don't totally understand what your trying to accomplish here. Conversion from a `datetime` object into an ISO timestamp should happen in the JSON encoder not in the model serializer. Am I missing something else here?

Reply:
In python3.6 I've seen the built in json encoder raise `TypeError: Object of type 'datetime' is not JSON serializable` when trying to send an embed that contains a datetime object but this could probably be solved with a custom json encoder model so i'll probably just pull that out of the pr for now.